rustls/src/crypto/ring/mod.rs

```diff
@@ -38,10 +45,17 @@ impl CryptoProvider for Ring {
         DEFAULT_CIPHER_SUITES
     }
 
-    /// Return all supported key exchange groups.
     fn default_kx_groups(&self) -> &'static [&'static dyn SupportedKxGroup] {
         ALL_KX_GROUPS
     }
+
+    fn load_private_signing_key(
+        &self,
+        key_der: PrivateKeyDer<'static>,
+    ) -> Result<Arc<dyn SigningKey>, Error> {
+        sign::any_supported_type(&key_der)
```

Comment:
Should we inline this? If we're exposing this, should we get rid of the `pub fn any_supported_type()` API?

(Should we hide `any_ecdsa_type()` and `any_eddsa_type()` functions from the public API? I forget why they are there in the first place.)

Reply:
I'm minded to leave these as they are, though if we were to replace this bit of the API I think it would look a bit different.

`any_ecdsa_type()` and co. is there to allow people to avoid linking in the whole of EdDSA and RSA if they know at compile time they will only use ECDSA keys. The ECDSA version also plasters over *ring*'s API which requires a-priori knowledge of which curve a key is on.